tests/test_frag.py

```diff
@@ -206,3 +228,77 @@ def test_symmetry(ext_args=FRAG_SYMMETRY):
     errors = []
     errors = td.check_file(os.getcwd(), "input.conf", PDB_lines, errors)
     assert not errors
+
+
+def test_fragment_atom(capsys, ext_args=FRAGMENT_ATOM):
+    """
+    Tests the frag_core_atom flag.
+    
+    Parameters
+    ----------
+    ext_args : str
+        Path to PELE input file.
+    """
+    if os.path.exists("input.conf"):
+        os.remove("input.conf")
+    try:
+        job = main.run_platform_from_yaml(ext_args)
+        captured = capsys.readouterr()
+        assert "Skipped - FragPELE will not run." not in captured.out
+
+    except Exception:
+        assert False
+
+
+def test_frag_waters(ext_args=FRAG_WATERS):
+    """
+    Check if water molecules are added to the system.
+  
+    Parameters
+    ----------
+    ext_args : str
+        Path to PELE input file,
+    """
+    water_output = []
+    job = main.run_platform_from_yaml(ext_args)
+    for path in job.working_dir:
+        output = glob.glob(os.path.join(path, "*_top.pdb"))[0]
+        with open(output, "r") as file:
+            lines = file.readlines()
+            for line in lines:
+                if line[17:21].strip() == "HOH":
+                    water_output.append(line.strip())
+
+    assert water_lines == water_output
+
+
+def test_ligand_clustering_production(ext_args=FRAG_CORE_ARGS):
+    """
+    Tests ligand clustering.
+
+    Parameters
+    ----------
+    ext_args : str
+        Path to PELE input file,
+    """
+    job = main.run_platform_from_yaml(ext_args)
+    for path in job.working_dir:
+        analysis_folder = os.path.join(path, "results")
+        if not os.path.isdir(analysis_folder):
+            assert False
+
+
+def test_ligand_clustering_libraries(ext_args=FRAG_SDF_LIBRARIES):
```

Comment:
Both those tests are doing exactly the same thing. Could we maybe do `@pytest.mark.parametrize`?